arrow-array/src/record_batch.rs

```diff
@@ -1637,4 +1653,57 @@ mod tests {
             "bar"
         );
     }
+
+    #[test]
+    fn test_batch_with_force_schema() {
+        fn force_schema_and_get_err_from_batch(
+            record_batch: &RecordBatch,
+            schema_ref: SchemaRef,
+            idx: usize,
+        ) -> Option<ArrowError> {
+            record_batch
+                .clone()
+                .with_schema_force(schema_ref)
+                .unwrap()
+                .project(&[idx])
+                .err()
+        }
+
+        let c: ArrayRef = Arc::new(StringArray::from(vec!["d", "e", "f"]));
+
+        let record_batch =
+            RecordBatch::try_from_iter(vec![("c", c.clone())]).expect("valid conversion");
+
+        // Test empty schema for non-empty schema batch
+        let invalid_schema_empty = Schema::empty();
+        assert_eq!(
+            force_schema_and_get_err_from_batch(&record_batch, invalid_schema_empty.into(), 0)
+                .unwrap()
+                .to_string(),
+            "Schema error: project index 0 out of bounds, max field 0"
+        );
+
+        // Wrong number of columns
+        let invalid_schema_more_cols = Schema::new(vec![
+            Field::new("a", DataType::Utf8, false),
+            Field::new("a", DataType::Int32, false),
```

Comment:
```suggestion
            Field::new("b", DataType::Int32, false),
```
?. This triggers my OCD 😅